app/web/src/components/IdeHeader/IdeHeader.tsx

```diff
@@ -92,89 +95,19 @@ const IdeHeader = ({
         <div />
       )}
       <div className="text-gray-200 grid grid-flow-col-dense gap-4 mr-4 items-center">
-        {canEdit && !projectTitle && (
-          <CaptureButton
+        {!children ? (
+          <DefaultTopButtons
+            project={project}
+            projectTitle={projectTitle}
+            _projectOwner={_projectOwner}
+            handleRender={handleRender}
             canEdit={canEdit}
-            projectTitle={project?.title}
-            userName={project?.user?.userName}
-            shouldUpdateImage={!project?.mainImage}
-            TheButton={({ onClick }) => (
-              <TopButton
-                onClick={onClick}
-                name="Save Project Image"
-                className=" bg-ch-blue-650 bg-opacity-30 hover:bg-opacity-80 text-ch-gray-300"
-              >
-                <Svg name="camera" className="w-6 h-6 text-ch-blue-400" />
-              </TopButton>
-            )}
           />
+        ) : (
+          children
```

Comment:
Not sure this is the right move but I made it so a component could override the default buttons by bringing their own as children to it. It let me remove them by providing a dummy element without effecting any other pages.

Reply:
Hmm I was trying to figure out what this even needed to be modified for the profile page but because this essentially the top nav bar.

I think this approach is good.

Perhaps later we'll look to remove `MainLayout.js` and replace it with the component everywhere. i.e. it can be the have the search bar for the projects explorer etc. If we do that than displaying children ONLY is probably better, i.e. it will need `DefaultTopButtons` to be passed in, in this case.

This is good though.

Reply:
Yes I like that idea about `MainLayout`, it's feeling about ready to be taken out after a little refactor of this header to cover more use cases.